README.md

```diff
@@ -1,155 +1,236 @@
 # flutter_flux
 
-> A Dart app architecture library with uni-directional data flow inspired by [RefluxJS](https://github.com/reflux/refluxjs) and Facebook's [Flux](https://facebook.github.io/flux/).
+A Dart/[Flutter](https://flutter.io) app architecture library with uni-directional data flow inspired by [RefluxJS](https://github.com/reflux/refluxjs) and Facebook's [Flux](https://facebook.github.io/flux/). [Flux](https://facebook.github.io/flux/) is the complementary “model” framework also designed by Facebook. Flutter_flux is a reimplementation of Flux that uses idiomatic Dart and integrates cleanly with Flutter.
 
-- [**Overview**](#overview)
-- [**What's Included**](#whats-included)
-  - [**Action**](#action)
-  - [**Store**](#store)
-  - [**FluxComponent**](#fluxcomponent)
-- [**Examples**](#examples)
-- [**External Consumption**](#external-consumption)
+- [**Motivation**](#motivation)
+- [**Widgets dispatch Actions**](#1-widgets-dispatch-actions)
+- [**Stores respond to dispatched Actions**](#2-stores-respond-to-dispatched-actions)
+- [**Widgets rebuild**](#3-widgets-rebuild)
+- [**FAQs**](#faqs)
 
 ---
 
-## Overview
+## Motivation
 
-![flux-diagram](https://github.com/Workiva/w_flux/blob/images/images/flux_diagram.png)
+The easiest ways to model state in Flutter are StatefulWidgets and global variables. StatefulWidgets are problematic because the data is destroyed whenever the widget hierarchy is destroyed. Therefore, state is lost when the route changes, and stateful widgets cannot be used for multiple views onto the same data.
 
-`flutter_flux` implements a uni-directional data flow pattern comprised of `Actions`, `Stores`, and `StoreWatchers`.
-It is based on [w_flux](https://github.com/Workiva/w_flux), but modified to use Flutter instead of React.
+Global variables also have issues:
+*   It’s necessary to handcraft the scaffolding to connect the global variables to the widget tree, often creating competing and/or incompatible implementations of the data model.
+*   There’s no data flow model, which is needed to make it easy to reason about the effects of dynamic data changing in different parts of the application.
 
-- `Actions` initiate mutation of app data that resides in `Stores`.
-- Data mutations within `Stores` trigger re-rendering of app view (defined in `StoreWatcher`).
-- Flutter `Widgets` and other interaction sources dispatch `Actions` in response to user interaction.
-- and the cycle continues...
+## Design
 
----
+The Flux framework is designed to solve the problems above. The dataflow in flutter_flux looks like this.
```

Comment:
Isn't Flux an architectural pattern? Would it be accurate to say "Flux framework" here?

I might also use "flutter_flux library" (instead of framework)